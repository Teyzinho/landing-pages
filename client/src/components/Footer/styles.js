import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 1px solid ${theme.colors.mediumGray};

    & ${TextComponent} {
      font-size: ${theme.font.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }
  `}
`;
