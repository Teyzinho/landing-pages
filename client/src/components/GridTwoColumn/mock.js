export default {
  title: 'Grid Two Column',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis sed officia cupiditate! Labore, eaque libero praesentium odio laudantium consectetur non, cum sunt error harum unde beatae optio, minus dolores illum.',
  srcImg:
    'https://th.bing.com/th/id/R.7685667ae3d9c1f62666809fa1b7eff4?rik=RtNLVQmuOO3JPQ&pid=ImgRaw&r=0',
};
