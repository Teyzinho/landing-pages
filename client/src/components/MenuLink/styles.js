import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    display: block;
    text-decoration: none;
    position: relative;
    width: fit-content;
    font-size: ${theme.font.sizes.small};
    padding: ${theme.spacings.small};
    color: ${theme.colors.primary};

    &::after {
      content: '';
      position: absolute;
      bottom: 0.75rem;
      left: 50%;
      width: 0;
      height: 0.2rem;
      background: ${theme.colors.secondary};
      transition: all 0.3s ease-in-out;
    }

    &:hover::after {
      width: 50%;
      left: 25%;
    }
  `}
`;
