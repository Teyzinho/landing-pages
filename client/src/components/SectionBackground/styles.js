import styled, { css } from 'styled-components';

const backgroundActivate = (theme) => css`
  background: ${theme.colors.primary};
  color: ${theme.colors.white};
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primary};
    ${background && backgroundActivate(theme)}
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
