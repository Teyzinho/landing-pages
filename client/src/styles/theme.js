export const theme = {
  colors: {
    primary: '#0A1128',
    secondary: '#dc143c',
    white: '#FFFFFF',
  },
  font: {
    family: {
      primary: "'Open Sans', sans-serif",
      secondary: "'Montserrat', sans-serif",
    },
    sizes: {
      xsmall: '1.2rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  spacings: {},
  media: {
    lteMedium: '(max-width: 768px)',
  },
};
