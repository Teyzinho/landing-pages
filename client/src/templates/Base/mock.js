import links from '../../components/NavLinks/mock';

export default {
  links: links,
  logoData: {
    text: 'logo',
    link: '#',
  },
  footerHtml: '<p>Teste do footer</p>',
};
