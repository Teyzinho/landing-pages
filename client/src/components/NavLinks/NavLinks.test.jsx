import { screen } from '@testing-library/react';
import { NavLinks } from '.';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';
import { theme } from '../../styles/theme';

describe('<NavLinks />', () => {
  it('should render links', () => {
    renderTheme(<NavLinks links={mock} />);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });

  // it('should render links', async () => {
  //   const { getByText } = renderTheme(<NavLinks links={mock} />);
  //   expect(getByText('Link 1').parentElement).toHaveStyleRule(
  //     'flex-flow',
  //     'column wrap',
  //     {
  //       media: theme.media.lteMedium,
  //     },
  //   );
  // });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={mock} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
