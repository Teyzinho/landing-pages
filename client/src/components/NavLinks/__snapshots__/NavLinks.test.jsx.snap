// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<NavLinks /> > should match snapshot 1`] = `
.c0 {
  display: flex;
  flex-flow: row wrap;
}

.c1 {
  display: block;
  text-decoration: none;
  position: relative;
  width: fit-content;
  font-size: 1.6rem;
  padding: 1.6rem;
  color: #0A1128;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0.75rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #dc143c;
  transition: all 0.3s ease-in-out;
}

.c1:hover::after {
  width: 50%;
  left: 25%;
}

@media (max-width: 768px) {
  .c0 {
    flex-flow: column wrap;
    align-content: center;
  }
}

<div>
  <nav
    class="c0"
  >
    <a
      class="c1"
      href="#link1"
      target="_self"
    >
      Link 1
    </a>
    <a
      class="c1"
      href="#link2"
      target="_self"
    >
      Link 2
    </a>
    <a
      class="c1"
      href="#link3"
      target="_self"
    >
      Link 3
    </a>
    <a
      class="c1"
      href="#link4"
      target="_self"
    >
      Link 4
    </a>
    <a
      class="c1"
      href="#link5"
      target="_self"
    >
      Link 5
    </a>
    <a
      class="c1"
      href="#link6"
      target="_self"
    >
      Link 6
    </a>
    <a
      class="c1"
      href="#link7"
      target="_self"
    >
      Link 7
    </a>
    <a
      class="c1"
      href="#link8"
      target="_self"
    >
      Link 8
    </a>
  </nav>
</div>
`;
